integration-tests/tests/src/tests/observable.ts

```diff
@@ -0,0 +1,848 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+/**
+ * This suite aims to test observability of Realm, Realm.Object and all collection types.
```

Comment:
Bit nit picky, but "observable" to is a term from RxJS and means something different when I read that.  Without reading this PR, I'm not sure I would deduce that event handlers were being tested here.

Reply:
I choose "observable" as a reference to the very general ["observer pattern"](https://en.wikipedia.org/wiki/Observer_pattern) in software design: The ability to register listeners (i.e. observers) of an object and have notifications trigger on change, which is probably also what RxJS and lots of other reactive / observable libraries are referencing.